src/transformers/models/emformer/processing_emformer.py

```diff
@@ -0,0 +1,106 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Speech processor class for Emformer
+"""
+import warnings
+from contextlib import contextmanager
+
+from ...processing_utils import ProcessorMixin
+from .feature_extraction_emformer import EmformerFeatureExtractor
+from .tokenization_emformer import EmformerTokenizer
+
+
+class EmformerProcessor(ProcessorMixin):
+    r"""
+    Constructs an Emformer processor which wraps an Emformer feature extractor and an Emformer target tokenizer into a
+    single processor.
+
+    [`EmformerProcessor`] offers all the functionalities of [`EmformerFeatureExtractor`] and [`PreTrainedTokenizer`].
+    See the docstring of [`~EmformerProcessor.__call__`] and [`~EmformerProcessor.decode`] for more information.
+
+    Args:
+        feature_extractor (`EmformerFeatureExtractor`):
+            An instance of [`EmformerFeatureExtractor`]. The feature extractor is a required input.
+        tokenizer ([`PreTrainedTokenizer`]):
+            An instance of [`PreTrainedTokenizer`]. The tokenizer is a required input.
+    """
+    feature_extractor_class = "EmformerFeatureExtractor"
+    tokenizer_class = "AutoTokenizer"
+
+    def __init__(self, feature_extractor, tokenizer):
+        super().__init__(feature_extractor, tokenizer)
+        self.current_processor = self.feature_extractor
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
+        try:
```

Comment:
Can we just have the following lines here:

```

            feature_extractor = EmformerFeatureExtractor.from_pretrained(pretrained_model_name_or_path, **kwargs)
            tokenizer = EmformerTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
```

and delete the rest? That was a bad leftover from Wav2Vec2